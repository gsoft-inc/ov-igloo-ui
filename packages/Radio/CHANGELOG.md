# @igloo-ui/radio

## 0.1.15

### Patch Changes

- 7d36815: Updated hopper token versions and igloo icon versions.

## 0.1.14

### Patch Changes

- a2df55d: Updated to latest Hopper Token dependency

## 0.1.13

### Patch Changes

- 812546b: fix the problem of displaying the hover of a selected radio

## 0.1.12

### Patch Changes

- 9739155: Gave ColorPicker the new Wokleap look. Color now uses css variables to update its color. Fixed spacing issues in tooltip. Updated hopper token version.

## 0.1.11

### Patch Changes

- c226be8: Applied the new style to the component.

## 0.1.10

### Patch Changes

- bf4ddef: Updated storybook version and linting. Made linting fixes in these components.

## 0.1.9

### Patch Changes

- a41e1d4: Fix: moved destructuring of props to function parameters so that storybook will automatically show the default value in the props table. It was done in Modal because it needed to send props to useDialog. In this case, @default annotation was used in comments.

## 0.1.8

### Patch Changes

- 6ee9c45: Updated anything causing errors or warnings in the browser console

## 0.1.7

### Patch Changes

- 0640409: udpate classnames to 2.3.2

## 0.1.6

### Patch Changes

- d971143: All tests were updated to use testing library and not enzyme

## 0.1.5

### Patch Changes

- 0a86f13: Update content of README

## 0.1.4

### Patch Changes

- 3d10593: fix position of radio button with long content and the size in small space

## 0.1.3

### Patch Changes

- 8c8686e: Change @import of font.css file for a @use in SASS files

## 0.1.2

### Patch Changes

- dd88c10: fix error of onChange not working

## 0.1.1

### Patch Changes

- b08048d: change disabled status color

## 0.1.0

### Minor Changes

- dd1d3c3: Initial release
