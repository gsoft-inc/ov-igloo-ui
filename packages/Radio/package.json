{
  "name": "@igloo-ui/radio",
  "version": "0.2.0",
  "main": "dist/Radio.js",
  "module": "dist/Radio.js",
  "types": "dist/Radio.d.ts",
  "styles": "dist/radio.css",
  "publishConfig": {
    "access": "public"
  },
  "repository": "git+https://github.com/gsoft-inc/ov-igloo-ui.git",
  "author": "Officevibe Inc.",
  "contributors": [
    "Franck franck.gaudin@gsoft.com"
  ],
  "license": "Apache-2.0",
  "peerDependencies": {
    "react": ">= 16.8.6"
  },
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c rollup.config.js"
  },
  "dependencies": {
    "@igloo-ui/tokens": "^2.0.0",
    "@hopper-ui/tokens": "^3.1.4",
    "classnames": "^2.3.2"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "IE >= 11"
  ]
}
