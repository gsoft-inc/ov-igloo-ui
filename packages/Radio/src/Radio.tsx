import * as React from "react";
import cx from "classnames";

import "./radio.scss";

export interface RadioProps extends React.ComponentPropsWithRef<"input"> {
    /** The content to display inside the label */
    children?: React.ReactNode;
    /** Add a specific class to the checkbox */
    className?: string;
    /** Add a data-test tag for automated tests */
    dataTest?: string;
    /** Indicates the ID of the element that is controlled by the checkbox */
    htmlFor: string;
    /** Changes the size of label  */
    small?: boolean;
    /** Modifies true/false value of the native checkbox */
    checked?: boolean;
    /** Modifies the native disabled state of the native checkbox */
    disabled?: boolean;
    /** The content to display to help users */
    helperText?: string;
    /** Function called when the value changes */
    onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

const Radio: React.FunctionComponent<RadioProps> = React.forwardRef(
    (
        {
            children,
            className,
            dataTest,
            htmlFor,
            onChange,
            helperText,
            small,
            checked,
            disabled,
            ...rest
        }: RadioProps,
        ref: React.Ref<HTMLInputElement>
    ) => {
        const classes = cx("ids-radio", className, {
            "ids-radio--small": small
        });

        return (
            <span
                className={cx("ids-form-control", {
                    "has-description": helperText
                })}
            >
                <input
                    ref={ref}
                    id={htmlFor}
                    className={classes}
                    data-test={dataTest}
                    checked={checked}
                    disabled={disabled}
                    type="radio"
                    onChange={onChange}
                    {...rest}
                />
                {children && (
                    <label className="ids-radio__label" htmlFor={htmlFor}>
                        {children}
                        {helperText && (
                            <span className="ids-radio__description">{helperText}</span>
                        )}
                    </label>
                )}
            </span>
        );
    }
);

export default Radio;
