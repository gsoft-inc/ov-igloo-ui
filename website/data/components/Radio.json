[
  [
    {
      "tags": {},
      "filePath": "/Users/franck.gaudin/Devel/Igloo/components/packages/Radio/src/Radio.tsx",
      "description": "",
      "displayName": "Radio",
      "methods": [],
      "props": {
        "children": {
          "defaultValue": null,
          "description": "The content to display inside the label",
          "name": "children",
          "parent": {
            "fileName": "components/packages/Radio/src/Radio.tsx",
            "name": "RadioProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Radio/src/Radio.tsx",
              "name": "RadioProps"
            },
            {
              "fileName": "components/node_modules/@types/react/index.d.ts",
              "name": "TypeLiteral"
            }
          ],
          "required": false,
          "type": { "name": "ReactNode" }
        },
        "className": {
          "defaultValue": null,
          "description": "Add a specific class to the checkbox",
          "name": "className",
          "parent": {
            "fileName": "components/packages/Radio/src/Radio.tsx",
            "name": "RadioProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Radio/src/Radio.tsx",
              "name": "RadioProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "dataTest": {
          "defaultValue": null,
          "description": "Add a data-test tag for automated tests",
          "name": "dataTest",
          "parent": {
            "fileName": "components/packages/Radio/src/Radio.tsx",
            "name": "RadioProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Radio/src/Radio.tsx",
              "name": "RadioProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "htmlFor": {
          "defaultValue": null,
          "description": "Indicates the ID of the element that is controlled by the checkbox",
          "name": "htmlFor",
          "parent": {
            "fileName": "components/packages/Radio/src/Radio.tsx",
            "name": "RadioProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Radio/src/Radio.tsx",
              "name": "RadioProps"
            }
          ],
          "required": true,
          "type": { "name": "string" }
        },
        "small": {
          "defaultValue": null,
          "description": "Changes the size of label",
          "name": "small",
          "parent": {
            "fileName": "components/packages/Radio/src/Radio.tsx",
            "name": "RadioProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Radio/src/Radio.tsx",
              "name": "RadioProps"
            }
          ],
          "required": false,
          "type": { "name": "boolean" }
        },
        "checked": {
          "defaultValue": null,
          "description": "Modifies true/false value of the native checkbox",
          "name": "checked",
          "parent": {
            "fileName": "components/packages/Radio/src/Radio.tsx",
            "name": "RadioProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Radio/src/Radio.tsx",
              "name": "RadioProps"
            }
          ],
          "required": false,
          "type": { "name": "boolean" }
        },
        "disabled": {
          "defaultValue": null,
          "description": "Modifies the native disabled state of the native checkbox",
          "name": "disabled",
          "parent": {
            "fileName": "components/packages/Radio/src/Radio.tsx",
            "name": "RadioProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Radio/src/Radio.tsx",
              "name": "RadioProps"
            }
          ],
          "required": false,
          "type": { "name": "boolean" }
        },
        "helperText": {
          "defaultValue": null,
          "description": "The content to display to help users",
          "name": "helperText",
          "parent": {
            "fileName": "components/packages/Radio/src/Radio.tsx",
            "name": "RadioProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Radio/src/Radio.tsx",
              "name": "RadioProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "onChange": {
          "defaultValue": null,
          "description": "Function called when the value changes",
          "name": "onChange",
          "parent": {
            "fileName": "components/packages/Radio/src/Radio.tsx",
            "name": "RadioProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Radio/src/Radio.tsx",
              "name": "RadioProps"
            }
          ],
          "required": false,
          "type": { "name": "(e: ChangeEvent<HTMLInputElement>) => void" }
        }
      }
    }
  ]
]
