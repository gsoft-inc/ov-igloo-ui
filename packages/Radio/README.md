# Radio

Use Radio components to be part of an options list where users can only make a single choice.

<Example />

<ReferenceLinks />

## Installation

To install `@igloo-ui/radio` in your project, you will need to run the following command using [npm](https://www.npmjs.com/):

```bash
npm install @igloo-ui/radio
```

If you prefer [Yarn](https://classic.yarnpkg.com/en/), use the following command instead:

```bash
yarn add @igloo-ui/radio
```

Then to use the component in your code just import it!

```js
import Radio from '@igloo-ui/radio';

<Radio htmlFor="radio-id">Label</Radio>;
```
