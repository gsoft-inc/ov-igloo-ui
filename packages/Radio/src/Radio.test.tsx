/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';

import Radio from './Radio';

const setUp = (props = {}) => {
  const component = shallow(<Radio htmlFor="radio-master" {...props} />);
  return component;
};

describe('Radio', () => {
  let component: any;
  beforeEach(() => {
    component = setUp();
  });

  test('It should render without errors', () => {
    const { getByText } = render(<Radio htmlFor="radio-test">Label</Radio>);
    expect(getByText('Label')).toBeInTheDocument();
  });

  test('It should render without label', () => {
    const { getByRole } = render(<Radio htmlFor="radio-test" />);
    expect(getByRole('radio')).toBeInTheDocument();
  });

  test('It should render a checked state', () => {
    const radio = setUp({ checked: true });
    expect(radio).toMatchSnapshot();
  });

  test('It should render a disabled state', () => {
    const radio = setUp({ disabled: true });
    expect(radio).toMatchSnapshot();
  });
});
