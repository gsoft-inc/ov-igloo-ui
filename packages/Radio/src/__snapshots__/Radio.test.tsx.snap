// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio It should render a checked state 1`] = `ShallowWrapper {}`;

exports[`Radio It should render a disabled state 1`] = `ShallowWrapper {}`;
